packages/editor/src/components/document-bar/index.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __, isRTL } from '@wordpress/i18n';
+import { useSelect, useDispatch } from '@wordpress/data';
+import {
+	Button,
+	VisuallyHidden,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+} from '@wordpress/components';
+import { BlockIcon } from '@wordpress/block-editor';
+import {
+	chevronLeftSmall,
+	chevronRightSmall,
+	page as pageIcon,
+	navigation as navigationIcon,
+	symbol,
+} from '@wordpress/icons';
+import { displayShortcut } from '@wordpress/keycodes';
+import { useEntityRecord } from '@wordpress/core-data';
+import { store as commandsStore } from '@wordpress/commands';
+import { useState, useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+const typeLabels = {
+	wp_pattern: __( 'Editing pattern:' ),
+	wp_navigation: __( 'Editing navigation menu:' ),
+	wp_template: __( 'Editing template:' ),
+	wp_template_part: __( 'Editing template part:' ),
```

Comment:
Should probably include a space at the end of the string so we can ensure the visually hidden text isn't stuck to the visible text; that's how the post editor does it. If there's no space it will sound weird on a screen reader.

Reply:
This whole thing is messed up to be honest: We should rely on concatenation of strings because depending on the language, it might not make sense. That said, it was that way before, so I'll see what I can but probably won't change much for now.